logins-api/android/library/src/test/java/org/mozilla/sync15/logins/LoginsStorageTest.kt

```diff
@@ -0,0 +1,340 @@
+/* Any copyright is dedicated to the Public Domain.
+   http://creativecommons.org/publicdomain/zero/1.0/ */
+
+package org.mozilla.sync15.logins
+
+import org.junit.Test
+import org.junit.Assert.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
+
+abstract class LoginsStorageTest {
+
+    abstract fun createTestStore(): LoginsStorage
+
+    private val encryptionKey = "testEncryptionKey"
+
+    private fun getTestStore(): LoginsStorage {
+        val store = createTestStore()
+
+        waitForResult(store.unlock(encryptionKey))
+
+        waitForResult(store.add(ServerPassword(
+            id = "aaaaaaaaaaaa",
+            hostname = "https://www.example.com",
+            httpRealm = "Something",
+            username = "Foobar2000",
+            password = "hunter2",
+            usernameField = "users_name",
+            passwordField = "users_password"
+        )))
+
+        waitForResult(store.add(ServerPassword(
+            id = "bbbbbbbbbbbb",
+            hostname = "https://www.example.org",
+            formSubmitURL = "https://www.example.org/login",
+            password = "MyVeryCoolPassword"
+        )))
+
+        waitForResult(store.lock())
+        return store
+    }
+
+    // TODO This should return T and not T?, I can't figure out how to coerce a T? to a T that
+    // doesn't throw if T is really nullable (e.g. `waitForResult<Int?>(SyncResult.fromValue(null))`
+    // should be fine!).
+    private fun <T> waitForResult(e: SyncResult<T>): T? {
```

Comment:
Is this cribbed from Mentat, or a derivative of Mentat?  This is not a good approach at all, and was straight up broken in Mentat: https://github.com/mozilla/mentat/commit/190e05e360e7d116d194a5aa2226cee717f3eaf9.

It's the test harness's responsibility to time out, so we shouldn't be baking timeouts in anywhere here.

Reply:
Hmm, neither actually, I didn't know mentat was doing the same thing!